Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -65,7 +65,8 @@ struct PathHierarchy {
     ) {
         var roots: [String: Node] = [:]
         var allNodes: [String: [Node]] = [:]
-        
+        var sparseNodes: [[String]: Node] = [:]
```

Comment:
Sparse nodes are only intended to be a last resort (and really only when `knownDisambiguatedPathComponents` is set but that's a separate issue that I plan on addressing). Would it be possible to fill in sparse nodes after all of the symbol graphs have been processed?

Reply:
If that's not possible. Please add FIXME comments in the code so that we can revisit this later.

Reply:
How would you propose handing the "orphan nodes" during symbol graph processing? I can poke at how i would want to handle it but having these nodes outside the hierarchy also feels like a dangerous situation even in the middle of initialization.

Reply:
I would suggest moving the existing code that processes symbols that don't belong in the hierarchy after all the symbol graph files have been processed, similar to how the overloads are processed. Best case, another symbol graph has a relationship for that symbol and the later processing is not needed. Worst case, nothing else references that symbol and we still have to find a location for it in the hierarchy (only with a more complete hierarchy than before).

Reply:
I made a mistake years ago by not restricting the processing of "orphan nodes" to _actual_ sparse symbol graph files like what the Convert Service uses. However, we can still accomplish that by gradually phasing out that logic in cases where it shouldn't happen until.